### PR TITLE
Blog App -validations and model specs.

### PR DESCRIPTION
**Pull request by Fombi-Favour**
In this activity, we: 
- added validations on User and Post models
- added unit tests for the methods and validations

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
```
Comment (Lucas-Masaba) [inline spec/models/user_spec.rb:1]: - [ ] Custom methods in the models should have corresponding unit tests. Please test the `recent_post` method in your User model

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
```
Comment (Lucas-Masaba) [inline spec/models/post_spec.rb:1]: - [ ] Custom methods in the models should have corresponding unit tests. Please test the `recent_comments` method in your Post model

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
```
Comment (Lucas-Masaba) [inline spec/models/like_spec.rb:1]: - [ ] Custom methods in the models should have corresponding unit tests. Please test the `like_counter_update` method in your Like model

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
```
Comment (Lucas-Masaba) [inline spec/models/comment_spec.rb:1]: - [ ] Custom methods in the models should have corresponding unit tests. Please test the `comment_counter_update` method in your Comment model

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Fombi-Favour  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Validations working :+1: 

## Required Changes ♻️
_Check the comments under the review._
- [ ] 8 out of 10 of your written tests are failing as seen below. Please make sure all your written tests are passing.
image


## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb
Comment (Efiamotu-1) [inline spec/models/post_spec.rb:1]: - [ ] Good job fixing the tests that were not working as requested by the previous reviewer however you are yet to add unit tests for the methods created in the post model as requested by the previous reviewer in this [review link](https://github.com/Fombi-Favour/blog-app-rb/pull/3#discussion_r1449395161), please kindly add them and resubmit for approval :+1: :sparkles:

File: spec/models/user_spec.rb
Comment (Efiamotu-1) [inline spec/models/user_spec.rb:1]: - [ ] Good job fixing the tests that were not working as requested by the previous reviewer however you are yet to add unit tests for the methods created in the user model as requested by the previous reviewer in this [review link](https://github.com/Fombi-Favour/blog-app-rb/pull/3#discussion_r1449394631), please kindly add them and resubmit for approval :+1: :sparkles:

File: spec/models/like_spec.rb
Comment (Efiamotu-1) [inline spec/models/like_spec.rb:1]: - [ ] Good job fixing the tests that were not working as requested by the previous reviewer however you are yet to add unit tests for the methods created in the like model as requested by the previous reviewer in this [review link](https://github.com/Fombi-Favour/blog-app-rb/pull/3#discussion_r1449395928), please kindly add them and resubmit for approval :+1: :sparkles:

File: spec/models/comment_spec.rb
Comment (Efiamotu-1) [inline spec/models/comment_spec.rb:1]: - [ ] Good job fixing the tests that were not working as requested by the previous reviewer however you are yet to add unit tests for the methods created in the comment model as requested by the previous reviewer in this [review link](https://github.com/Fombi-Favour/blog-app-rb/pull/3#discussion_r1449397541), please kindly add them and resubmit for approval :+1: :sparkles:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @Fombi-Favour , 



Great work on making the changes requested by a previous reviewer 👏 Your Code is Awesome! 💪
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    end
+
+    it 'likes_counter should be greater than or equal to zero' do
+      post1 = Post.new(likes_counter: -1)
+      post1.valid?
+      expect(post1.errors[:likes_counter]).to include('must be greater than or equal to 0')
+    end
+
+    it 'post_counter should update' do
+      initial = User.find(1).post_counter
```
Comment (zieeco) [inline spec/models/post_spec.rb]: image
---
- Your test is failing, please, see the image for the detailed error to fix it.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Fombi-Favour,





Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Nice effort making the requested changes. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 5 above)
Comment (Fombi-Favour) [reply spec/models/post_spec.rb]: I ran the test code and its a pass. Please consider my request 🙏 
Screenshot 2024-01-13 020348

**Review by shahadat3669 — APPROVED** [review]
Hi @Fombi-Favour, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Worked on previous reviewer comments.
✅ All test cases passed.


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**